backend/tests/test_api.py

```diff
@@ -1,42 +1,43 @@
 import pytest
-from backend.editor.api import api
-from falcon import testing
 
 
 @pytest.fixture(autouse=True)
 def test_setup(neo4j):
     # delete all the nodes and relations in the database
-    query = "MATCH n DETACH DELETE n"
+    query = "MATCH (n) DETACH DELETE n"
     neo4j.session().run(query)
     query = "DROP INDEX p_test_branch_SearchIds IF EXISTS"
     neo4j.session().run(query)
     query = "DROP INDEX p_test_branch_SearchTags IF EXISTS"
     neo4j.session().run(query)
 
-@pytest.fixture()
-def client():
-    return testing.TestClient(api)
 
 def test_hello(client):
     response = client.get("/")
     assert response.status_code == 200
     assert response.json() == {"message": "Hello user! Tip: open /docs or /redoc for documentation"}
 
-def delete_project_test(neo4j):
-    session = neo4j.session()
-    query = "MATCH (n:p_test_branch:t_test:b_branch:ENTRY) RETURN COUNT(*)"
 
+def test_delete_project(neo4j, client):
+    session = neo4j.session()
     create_project = """
-            CREATE (n:PROJECT)
-            SET n.id = test_project
-            SET n.taxonomy_name = testtaxonomyname
-            SET n.branch_name = testbranchname
-            SET n.description = testdescription
-            SET n.status = OPEN
-            SET n.created_at = datetime()
-        """
+        CREATE (n:PROJECT)
+        SET n.id = 'test_project'
+        SET n.taxonomy_name = 'test_taxonomy_name'
+        SET n.branch_name = 'test_branch_name'
+        SET n.description = 'test_description'
+        SET n.status = 'OPEN'
+        SET n.created_at = datetime()
+    """
     result = session.run(create_project)
-    import pdb; pdb.set_trace();
-    stimulate_api_result = client.simulate_get("/testtaxonomyname/testbranchname/delete")
-    assert stimulate_api_result.status_code == 200
-    assert stimulate_api_result.json == {}
+    response = client.delete("/test_taxonomy_name/test_branch_name/delete")
+    assert response.status_code == 200
+    # assert response.json() == {"message": ""}
```

Comment:
Though this test case passes, it does not completed the test of the API.

After invoking the delete API I do not get the count of number of nodes or relations delete. I tried this in Neo4j browser as well.

The query is straightforward. Which makes me suspect that my data is incomplete here. 

What nodes should I create for this test to be successful?

Reply:
you didn't return anything in you method, that's why !